### PR TITLE
Date range/decade filters

### PR DESCRIPTION
**Pull request by benvogler**
image
image
image

### CODE REVIEW

File: client/src/components/Search/Filters/Filters.js [hunk 1]
```diff
@@ -1,26 +1,46 @@
 import * as React from 'react';
 import './Filters.scss';
 import { withRouter } from 'react-router-dom';
-import { Form } from 'react-bootstrap';
+import { Collapse, Form } from 'react-bootstrap';
 import TagsSearch from '../TagsSearch/TagsSearch';
 import search from "../../../utils/search";
+import DatePicker from 'react-datepicker/es';
```
Comment (njd9419) [inline client/src/components/Search/Filters/Filters.js]: What's the difference of importing `react-datepicker/es` and 'react-datepicker'?

Comment (njd9419) [reply client/src/components/Search/Filters/Filters.js]: Espanol senior.

File: client/src/components/Search/Filters/Filters.js [hunk 2]
```diff
@@ -1,26 +1,46 @@
...
 class Filters extends React.Component {
 
     constructor(props) {
         super(props);
-        const { distance } = props;
-        this.state = { distance };
+        const { distance, decade, start, end } = props;
+        let dateFiltersMode = 'hidden';
+        if (decade) dateFiltersMode = 'decade';
+        if (start && end) dateFiltersMode = 'range';
```
Comment (njd9419) [inline client/src/components/Search/Filters/Filters.js]: Could be `else if...`?

File: src/main/java/com/monumental/services/MonumentService.java [hunk 3]
```diff
@@ -247,13 +253,17 @@ private void buildSearchQuery(CriteriaBuilder builder, CriteriaQuery query, Root
      * @return List<Monument> - List of Monument results based on the specified search parameters
```
Comment (njd9419) [inline src/main/java/com/monumental/services/MonumentService.java]: Can you add the new params to this block comment?

File: src/main/java/com/monumental/services/MonumentService.java [hunk 4]
```diff
@@ -182,7 +182,7 @@ private Predicate buildTagsQuery(CriteriaBuilder builder, CriteriaQuery query, R
      */
```
Comment (njd9419) [inline src/main/java/com/monumental/services/MonumentService.java]: Can you add the new params to this block comment?

**Comment by njd9419** [comment]
Looks good. Just a request for a few additional comments and some minor things.

**Comment by benvogler** [comment]
@njd9419 I couldn't remember what changes Dr. Decker wanted with this before break, so I've update it to master and will just merge it as-is and get her thoughts on it thursday (or you guys will have to if I'm not there)

**Comment by njd9419** [comment]
Sounds good.

**Pull request merged**